pkg/k8s/network_policy.go

```diff
@@ -185,6 +185,19 @@ func ParseNetworkPolicy(np *networkingv1.NetworkPolicy) (api.Rules, error) {
 			egress.ToEndpoints = append(egress.ToEndpoints, all)
 		}
 
+		if eRule.Ports != nil && len(eRule.Ports) > 0 {
+			egress.ToPorts = parsePorts(eRule.Ports)
+		} else if eRule.To == nil || len(eRule.To) == 0 {
```

Comment:
I like this being explicit but it can be reduced to:
```golang
if len(eRule.Ports) > 0 {

} else {

}
```
since the len will treat nil as an empty as well.
On the other hand, I like the explicit handling of nil because it highlights the logic around the parsing. It might be worth moving the comment below to just above this block to provide context for these ifs.

Reply:
Interestingly enough, removing the nil checks broke unit tests, so I didn't add this. I think we need this to differentiate between nil and empty slices.